app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -50,23 +51,23 @@ internal fun normalModeAdapterItems(
         items.add(AdapterItem.ExperimentDefaultBrowserCard)
     }
 
-    if (topSites.isNotEmpty()) {
+    if (settings.showTopSitesFeature && topSites.isNotEmpty()) {
```

Comment:
This is because we don't clear the state when we turn off the setting if it was enable prior. This was highlighted in https://github.com/mozilla-mobile/fenix/pull/24138#pullrequestreview-913246170

> Decide if we should keep this state up-to-date even if the home fragment is not displayed